### ISSUE TITLE
[Bug]: a11y: LLM Provider selection with a screenreader not possible anymore in Firefox since v0.7.6

### ISSUE THREAD

**Issue by fcnjd** [danny-avila/LibreChat#5119]
### What happened?

Until the latest update, I could always expand the button with the LLM provider, and then change it using up and down arrow keys (in this case I used the virtual buffer/browse mode of my screenreader, not the usual keyboard navigation). With the latest update to v0.7.6, this functionality breaks. It looks like now you have to use tab key, after the selection is expanded, to change the provider. Even if this works on Chrome, this doesn't work on Firefox, NVDA screenreader won't read out which entry from the list is currently selected. It would be great if this could get working again.

### Steps to Reproduce

1. Turn on NVDA screenreader and use Firefox
2. Activate the LLM provider selection
3. press tab to cycle through the entries, nothing is spoken

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thank you for pointing this out. I can see that NVDA is reading "section" in Firefox for each option. I will also provided support for Up/down arrows.

The changes here were made to better conform to WCAG for the HTML structure of this dropdown list, however, I apologize this was not fully tested across browsers
